libs/domain/user/address-remove/address-remove.component.ts

```diff
@@ -1,56 +1,79 @@
 import { ButtonType } from '@spryker-oryx/ui/button';
 import { IconTypes } from '@spryker-oryx/ui/icon';
-import { Address, AddressMixin } from '@spryker-oryx/user';
+import { AddressMixin } from '@spryker-oryx/user';
 import { hydratable, i18n, Size } from '@spryker-oryx/utilities';
 import { html, LitElement, TemplateResult } from 'lit';
-import { CANCEL_EVENT, CONFIRM_EVENT } from './address-remove.model';
+import { state } from 'lit/decorators.js';
+import { tap } from 'rxjs';
 import { styles } from './address-remove.styles';
 
 @hydratable(['mouseover', 'focusin'])
 export class UserAddressRemoveComponent extends AddressMixin(LitElement) {
   static styles = styles;
 
-  protected override render(): TemplateResult | void {
-    const address = this.$address();
-    if (!address) return;
+  @state() protected requestsConfirmation = false;
+  @state() protected loading = false;
 
+  protected override render(): TemplateResult | void {
     return html`
-      <oryx-user-address .addressId=${address.id}></oryx-user-address>
-      <section>
-        <oryx-icon .type=${IconTypes.Info} size=${Size.Md}></oryx-icon>
-        <span>
-          ${i18n(
-            'user.address.removing-this-address-will-not-remove-any-pending-orders-being-dispatched-to-this-address'
-          )}
-        </span>
-      </section>
-      ${this.renderControls(address)}
+      <oryx-icon-button>
+        <button
+          aria-label=${i18n('user.address.remove')}
+          @click=${this.onConfirm}
+        >
+          <oryx-icon .type=${IconTypes.Trash}></oryx-icon>
+        </button>
+      </oryx-icon-button>
+      ${this.renderRemoveConfirmationModal()}
     `;
   }
 
-  protected renderControls(address: Address): TemplateResult {
-    return html`
-      <oryx-button outline type=${ButtonType.Secondary}>
-        <button @click=${(): void => this.emitEvent(CANCEL_EVENT)}>
-          ${i18n('user.address.cancel')}
-        </button>
-      </oryx-button>
+  protected onConfirm(): void {
+    this.requestsConfirmation = true;
+  }
 
-      <oryx-button type=${ButtonType.Critical}>
-        <button @click=${(): void => this.emitEvent(CONFIRM_EVENT, address)}>
-          ${i18n('user.address.remove')}
-        </button>
+  protected renderRemoveConfirmationModal(): TemplateResult | void {
+    if (!this.requestsConfirmation) return;
+
+    return html`<oryx-modal
+      open
+      .heading=${i18n('checkout.address.remove-address')}
```

Comment:
```suggestion
      .heading=${i18n('oryx.checkout.address.remove-address')}
```

Should our translations have `oryx` prefix?

Reply:
no, we don't need that.